From f441cb9a70fa3f55e9bbd615924879d692d21a6c Mon Sep 17 00:00:00 2001
From: Florian Weimer <fweimer@redhat.com>
Date: Mon, 11 Sep 2023 09:17:52 +0200
Subject: [PATCH 112/118] elf: Move l_init_called_next to old place of
 l_text_end in link map

This preserves all member offsets and the GLIBC_PRIVATE ABI
for backporting.
---
 include/link.h | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/include/link.h b/include/link.h
index 81f5715db07535b7e5d437da92d23f197bd910f2..4eb8fe0d96e0be183c963285673c7b3c894659ca 100644
--- a/include/link.h
+++ b/include/link.h
@@ -254,6 +254,10 @@ struct link_map
        need not be the same as l_addr.  */
     ElfW(Addr) l_map_start, l_map_end;
 
+    /* Linked list of objects in reverse ELF constructor execution
+       order.  Head of list is stored in _dl_init_called_list.  */
+    struct link_map *l_init_called_next;
+
     /* Default array for 'l_scope'.  */
     struct r_scope_elem *l_scope_mem[4];
     /* Size of array allocated for 'l_scope'.  */
@@ -276,10 +280,6 @@ struct link_map
     /* List of object in order of the init and fini calls.  */
     struct link_map **l_initfini;
 
-    /* Linked list of objects in reverse ELF constructor execution
-       order.  Head of list is stored in _dl_init_called_list.  */
-    struct link_map *l_init_called_next;
-
     /* List of the dependencies introduced through symbol binding.  */
     struct link_map_reldeps
       {
-- 
2.41.0

